core/src/test/scala/unit/kafka/controller/ControllerEventManagerTest.scala

```diff
@@ -135,6 +135,44 @@ class ControllerEventManagerTest {
     assertEquals(500, queueTimeHistogram.max, 0.01)
   }
 
+  @Test
+  def testEventQueueTimeResetOnTimeout(): Unit = {
+    val metricName = "kafka.controller:type=ControllerEventManager,name=EventQueueTimeMs"
+    val controllerStats = new ControllerStats
+    val time = new MockTime()
+    val latch = new CountDownLatch(1)
+    val processedEvents = new AtomicInteger()
+
+    val eventProcessor = new ControllerEventProcessor {
+      override def process(event: ControllerEvent): Unit = {
+        latch.await()
+        time.sleep(500)
```

Comment:
I don't think we need this for this test. The test above has this because they want to check that the `max` is `500`.